docs/extend/modular-resources/examples/odrive.md

```diff
@@ -0,0 +1,281 @@
+
+{{% alert title="Tip" color="tip" %}}
+
+Navigate to the `odrive/odrive-motor` directory of the ODrive module you cloned on a Terminal session with your computer and run `pwd` to obtain the path to `run.sh` on your robot's computer.
+
+``` {id="terminal-prompt" class="command-line" data-prompt="$"}
+cd <your/path/to/odrive/odrive-motor>
+pwd
+```
+
+{{% /alert %}}
+
+## Configuration
+
+### Module
+
+{{< tabs name="Add the ODrive module">}}
+{{% tab name="Config Builder" %}}
+
+Go to your robot's page on the [Viam app](https://app.viam.com/).
+Navigate to the **Config** tab on your robot's page, and click on the **Modules** subtab.
+
+Add the ODrive module with a name of your choice and the executable path that points to the location where your ODrive module’s run script, <file>run.sh</file>, is stored on your robot’s computer.
+
+![The ODrive module with the name "odrive" and executable path "<your/path/to/run.sh>" added to a robot in the Viam app config builder](../../img/add-odrive/add-odrive-module-ui.png)
+
+{{% /tab %}}
+{{% tab name="JSON Template" %}}
+
+```json {class="line-numbers linkable-line-numbers"}
+"modules": [
+  {
+    "name": "odrive",
+    "executable_path": "<your/path/to/run.sh>"
+  }
+]
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### Modular Resource
+
+<!-- TODO sg: Unsure if this is supported in module registry stat
+
+name="Config Builder" 
+
+Go to your robot's page on the [Viam app](https://app.viam.com/).
+Navigate to the **Config** tab on your robot's page, and click on the **Components** subtab.
+
+Add a new component with a **Name** of your choice and the **Type** `motor`.
+
+In **Model**, enter either `viam:motor:odrive-serial` or `viam:motor:odrive-canbus`, depending on your preferred connection method.
+A warning tooltip will appear explaining this model is not built-in to the RDK, so it is not recognized by Viam and won't have validation features.
+
+![The ODrive motor component in the serial model being added as a component with name "your-odrive-serial" in the Viam app config builder](../../img/add-odrive/add-odrive-resource-ui.png)
+
+Click **Create component**.
+Edit and fill in the attributes as applicable to your model of ODrive.
+
+![The ODrive motor component in the serial model with example configuration attributes pasted in the Viam app config builder](../../img/add-odrive/add-odrive-attributes-ui.png)
+
+tab -->
+
+Go to your robot's page on the [Viam app](https://app.viam.com/).
+Navigate to the **Config** tab on your robot's page, and click on the **Components** subtab.
+Select **Raw JSON** mode.
+
+Copy and paste the following JSON depending on your preferred communication type:
+
+{{< tabs name="Add an ODrive motor">}}
+{{% tab name="odrive-serial" %}}
+
+```json {class="line-numbers linkable-line-numbers"}
+//  < INSERT YOUR ODRIVE MODULE CONFIG >
+  "components": [
+    {
+      "model": "viam:motor:odrive-serial",
+      "namespace": "rdk",
+      "attributes": {
```

Comment:
@npentrel I started on this and realized it might not be good as Rand and Martha have notes to discourage use of the optional attributes for the most part (excluding the use cases they're built for) other than the canbus for can. Specifying a serial number or config file path could actually make the odrive resource flakier for users if I understand correctly

Reply:
yeah it is kind of an interesting situation. for serial, there's really no harm in using the optional attributes such as serial number and config file, because the odrive has to be plugged in over usb for the serial implementation to work at all. basically, we want to make it clear that for the can model, ideally you would only use the can node id, and set everything up before hand so that the serial numbers and config files aren't necessary, but for the serial model it doesn't actually really matter